torch/_dynamo/variables/torch.py

```diff
@@ -494,6 +494,28 @@ def get_state_from_generator():
             return TorchVariable(torch.add, **options).call_function(
                 tx, [args[0], result], {}
             )
+        elif self.value == torch.utils._pytree.tree_flatten:
+            if len(args) != 1:
+                unimplemented("Unsupported flatten with len(args) != 1")
+
+            items = args[0].unpack_var_sequence(tx)
+            flattened, spec = torch.utils._pytree.tree_flatten(items)
+            return TupleVariable(
+                [ListVariable(flattened), ConstantVariable(spec)], **options
```

Comment:
Let's create a new variable type for a spec.

Reply:
I got pretty far without this, using pytree registrations, do you mind looking again and seeing if we need it?